lib/trie/node.go

```diff
@@ -211,71 +219,94 @@ func (b *branch) setKey(key []byte) {
 	b.key = key
 }
 
-func (b *branch) encodeAndHash() ([]byte, []byte, error) {
+func (b *branch) encodeAndHash() (encoding, hash []byte, err error) {
```

Comment:
why do we have both `encodeAndHash` functions on the package level and on the node receiver functions?

Reply:
This should be addressed to some extent in #2077